common/remote/hooks/git/gitfile.cpp

```diff
@@ -263,6 +263,15 @@ class GitRepositoryFile : implements IFile, public CInterface
     {
         if (!isExisting)
             return (offset_t) -1;
+        if (fileSize != (offset_t) -1)
+            return fileSize;
+        git_blob *blob = nullptr;
```

Comment:
```
if (isDir)
   return 0;

```
either here or set fileSize in the GitRepositoryFile constructor.

Reply:
Fixed and repushed